linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -126,14 +138,17 @@ private[emitter] final class KnowledgeGuardian(config: Emitter.Config) {
   }
 
   private def computeStaticFieldMirrors(
-      topLevelExports: List[LinkedTopLevelExport]): Map[ClassName, Map[FieldName, List[String]]] = {
+      moduleSet: ModuleSet): Map[ClassName, Map[FieldName, List[String]]] = {
     if (config.moduleKind != ModuleKind.NoModule) {
       Map.empty
     } else {
       var result = Map.empty[ClassName, Map[FieldName, List[String]]]
-      for (export <- topLevelExports) {
+      for {
+        module <- moduleSet.modules
+        export <- module.topLevelExports
+      } yield {
```

Comment:
The `yield` shouldn't be there, I think. This for comprehension is in statement position, so its result is ignored.
```suggestion
      } {
```